packages/jest-resolve/src/defaultResolver.ts

```diff
@@ -160,12 +162,25 @@ function getPathInModule(
       );
     }
 
-    if (resolved.startsWith('.')) {
-      return pathResolve(dirname(closestPackageJson), resolved);
+    const resolvedValues = Array.isArray(resolved) ? resolved : [resolved];
+
+    for (const resolved of resolvedValues) {
+      const resolvedPath = resolveByPath(resolved);
+      try {
+        return resolveSync(resolvedPath, options);
```

Comment:
why are we calling `resolve` here?

Reply:
To handle the array pattern.

Because in array pattern you need to actually resolve and see if the file/module exists

Reply:
Right now the test is failing. Need to see what is the problem.

Likely the resolution failed somehow:

```
● Validation Error:

  Module react-native should have "jest-preset.js" or "jest-preset.json" file at the root.

  Configuration Documentation:
  https://jestjs.io/docs/configuration
  ```

Reply:
why can we not call the old logic in a loop? i.e. if external module, resolve it, otherwise just create the file path. We can of course add a `isFile` check, but a full resolve seems overkill

Reply:
There is one case I should fix thou. That it should rethrow the last error.

Reply:
we also need to handle if none of the paths can be resolved - right now you check all, and if none match just return `undefined`. We should throw some sort of error

Reply:
Because the return type is `string | string[] | undefined`. So the array case needs to be handled within.

Another way is to externalize this code instead:

```ts
      if (resolved.startsWith('.')) {
        return pathResolve(dirname(closestPackageJson!), resolved);
      }

      // this is an external module, re-resolve it
      return defaultResolver(resolved, options);
```

So that it can be a loop in the main default resolver function.


> We should throw some sort of error

Yes. I'm planning to change it to rethrow the last error.

Reply:
>Another way is to externalize this code instead:

That _sounds_ cleaner

Reply:
Because the return type is `string | string[] | undefined`. So the array case needs to be handled within.

Another way is to externalize this code instead:

```ts
      if (resolved.startsWith('.')) {
        return pathResolve(dirname(closestPackageJson!), resolved);
      }

      // this is an external module, re-resolve it
      return defaultResolver(resolved, options);
```

So that it can be a loop in the main default resolver function.


> We should throw some sort of error

Yes. I'm planning to change it to rethrow the last error.

Reply:
ar, doesn't work. the `closestPackageJson` is inside the function

Reply:
With the array pattern out of the way, I'll be reverting this. Don't need this change anymore.